File: src/yt/item5/bean/Shoes.java
package yt.item5.bean;

import java.io.Serializable;

public class Shoes implements Serializable, EntityInterface {

	private static final long serialVersionUID = 1L;

	private int shoesId;

	private String shoesName;

	private String series;

	private String category;

	private int price;

	private Brand brand;

	public Shoes() {

	}

	public Shoes(String shoesName) {
		setShoesName(shoesName);
	}

	public int getShoesId() {
		return shoesId;
	}

	public Shoes setShoesId(int shoesId) {
		this.shoesId = shoesId;
		return this;
	}

	public String getShoesName() {
		return shoesName;
	}

	public Shoes setShoesName(String shoesName) {
		this.shoesName = shoesName;
		return this;
	}

	public String getSeries() {
		return series;
	}

	public Shoes setSeries(String series) {
		this.series = series;
		return this;
	}

	public String getCategory() {
		return category;
	}

	public Shoes setCategory(String category) {
		this.category = category;
		return this;
	}

	public int getPrice() {
		return price;
	}

	public Shoes setPrice(int price) {
		this.price = price;
		return this;
	}

	public Brand getBrand() {
		return brand;
	}

	public Shoes setBrand(Brand brand) {
		this.brand = brand;
		return this;
	}

	public Shoes setBrandById(int id) {
		this.brand = new Brand(id);
		return this;
	}

	@Override
	public int getId() {
		return this.shoesId;
	}

	@Override
	public String toString() {
		return shoesId + " " + shoesName + " " + series + " " + price + " " + brand;

	}
}

File: WebContent/js/jsModel/Shoes.js
function Shoes() {
	var shoesId, shoesName, series, category, price;

	// var brand = new Brand();

	this.setShoesId = function(param) {
		shoesId = param;
		return this;
	}
	this.setShoesName = function(param) {
		shoesName = param;
		return this;
	}
	this.setSeries = function(param) {
		series = param;
		return this;
	}
	this.setCategory = function(param) {
		category = param;
		return this;
	}
	this.setPrice = function(param) {
		price = param;
		return this;
	}
	this.setAttributeByObj = function(obj) {
		shoesId = obj['shoesId'];
		shoesName = obj['shoesName'];
		series = obj['series'];
		category = obj['category'];
		price = obj['price'];
		return this;
	}
	this.getAttributes = function() {
		return {
			'shoesId' : shoesId,
			'shoesName' : shoesName,
			'series' : series,
			'category' : category,
			'price' : price
		};
	}
	this.getShoesId = function() {
		return shoesId;
	}
	this.getShoesName = function() {
		return shoesName;
	}
	this.getSeries = function() {
		return series;
	}
	this.getCategory = function() {
		return category;
	}
	this.getPrice = function() {
		return price;
	}
};

Shoes.prototype = (function() {

	return {
		colAttribute : ['shoesId', 'shoesName', 'series', 'category', 'price'],
		entityServlet : 'ShoesTableController',
		inputColAttributeMap : {
			shoesId : {
				colTag : 'input',
				colType : 'text',
				otherAttribute : 'readOnly'
			},
			shoesName : {
				colTag : 'input',
				colType : 'text',
				otherAttribute : 'requierd'
			},
			series : {
				colTag : 'input',
				colType : 'text'
			},
			category : {
				colTag : 'input',
				colType : 'text'
			},
			price : {
				colTag : 'input',
				colType : 'number'
			}
		}

	}
}());